CRM/Core/Resources.php

```diff
@@ -257,12 +257,18 @@ public function addScript($code, $weight = self::DEFAULT_WEIGHT, $region = self:
    * @param string $nameSpace
    *   Usually the name of your extension.
    * @param array $vars
+   * @param string $region
+   *   The region to add settings to (eg. for payment processors usually billing-block
+   *
    * @return CRM_Core_Resources
    */
-  public function addVars($nameSpace, $vars) {
+  public function addVars($nameSpace, $vars, $region = NULL) {
+    if (!$region) {
+      $region = self::isAjaxMode() ? 'ajax-snippet' : 'html-header';
+    }
```

Comment:
In order to preserve existing functionality, these 3 lines need to be moved to the `addSetting()` function. Removing it from there will mess up anything that calls that function directly (it's a public function).

Reply:
D'oh! - fixed